src/context.cc

```diff
@@ -94,29 +94,45 @@ DeviceOrd CUDAOrdinal(DeviceOrd device, bool) {
   StringView msg{R"(Invalid argument for `device`. Expected to be one of the following:
 - cpu
 - cuda
-- cuda:<device ordinal>  # e.g. cuda:0
+- cuda:<device ordinal>            # e.g. cuda:0
 - gpu
-- gpu:<device ordinal>   # e.g. gpu:0
+- gpu:<device ordinal>             # e.g. gpu:0
+- sycl
+- sycl:<device ordinal>            # e.g. sycl:0
+- sycl:<cpu, gpu>
+- sycl:<cpu, gpu>:<device ordinal> # e.g. sycl:gpu:0
 )"};
   auto fatal = [&] { LOG(FATAL) << msg << "Got: `" << input << "`."; };
 
 #if defined(__MINGW32__)
   // mingw hangs on regex using rtools 430. Basic checks only.
   CHECK_GE(input.size(), 3) << msg;
   auto substr = input.substr(0, 3);
-  bool valid = substr == "cpu" || substr == "cud" || substr == "gpu";
+  bool valid = substr == "cpu" || substr == "cud" || substr == "gpu" || substr == "syc";
   CHECK(valid) << msg;
 #else
-  std::regex pattern{"gpu(:[0-9]+)?|cuda(:[0-9]+)?|cpu"};
+  std::regex pattern{"gpu(:[0-9]+)?|cuda(:[0-9]+)?|cpu|sycl(:cpu|:gpu)?(:-1|:[0-9]+)?"};
   if (!std::regex_match(input, pattern)) {
     fatal();
   }
 #endif  // defined(__MINGW32__)
 
   // handle alias
-  std::string s_device = std::regex_replace(input, std::regex{"gpu"}, DeviceSym::CUDA());
+  std::string s_device = input;
+  if (!std::regex_match(s_device, std::regex("sycl(:cpu|:gpu)?(:-1|:[0-9]+)?")))
+    s_device = std::regex_replace(s_device, std::regex{"gpu"}, DeviceSym::CUDA());
```

Comment:
Shouldn't this mean 
User specify any GPU on the system -> xgboost check available devices and pickup GPU device that i can execute on. 
I.e. in case there is only CUDA device - it will be used
In case there is only SYCL device - it will be used
The question what to do in case both are available.. Might be warning about non deterministic selection or might be even error. Not sure how realistic this case is

Reply:
I guess in the future it should work like you have described. But let's postpone this fix for future.